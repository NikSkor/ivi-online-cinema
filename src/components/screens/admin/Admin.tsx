import { FC } from "react";
import Layout from '@/components/layout/Layout';
import BreadCrumbs from '@/components/ui/breadCrumbs/BreadCrumbs';
import style from './Admin.module.scss';
import { filmographyData } from "@/components/ui/filmography/filmography.data";
import { IFilmographyItem } from "@/interfaces/person/IFilmographyItem";
import Image from 'next/image';
import trashImg from '../../../../public/trash.svg';
import Link from "next/link";

const Admin: FC = () => {

  let catalog: IFilmographyItem[] = [...filmographyData];

  return(
    <div className="container">
      <section className={style.header}>
        <BreadCrumbs 
          pathList={[{pathLink: '/', pathName: 'Мой Иви'}]} 
          slug={'Администратор'} /> 
      </section>
      <section className={style.main}>
        <h2 className={style.title}>Управление каталогом</h2>
        <h3 className={style.subTitle}>Список фильмов</h3>
        <ul className={style.list}>
        {catalog.map((item)=> {
          return <li key={item.filmId} className={style.item}>
              <h4 className={style.itemTitle}>{item.name}</h4>
              <Link href={`/admin/${item.filmId}`} className={style.actionBtn} >
                <p data-id={item.filmId}>Редактировать</p>
              </Link>
              <button className={style.actionBtn} onClick={()=>{}}>
                <Image src={trashImg} data-id={item.filmId} alt="Значок очистки"/>
              </button>
            </li>
        })}
      </ul>
        <button className={style.actionBtn}>
          Добавить
        </button>
      </section>
    </div>
  )
}

export default Admin;