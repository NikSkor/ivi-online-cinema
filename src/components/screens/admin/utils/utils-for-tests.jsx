import React from "react";
import { render } from "@testing-library/react";
import { configureStore } from "@reduxjs/toolkit";
import { Provider } from "react-redux";
import adminReducer from '../../../../store/slices/adminSlice';


export function renderWithProviders(
  ui,
  {
    preloadedState = {},
    store = configureStore({
      reducer: { admin: adminReducer},
      preloadedState,
    }),
    ...renderOptions
  } = {}
) {
  function Wrapper({ children }) {
    return <Provider store={store}>{children}</Provider>;
  }

  return { store, ...render(ui, { wrapper: Wrapper, ...renderOptions }) };
}