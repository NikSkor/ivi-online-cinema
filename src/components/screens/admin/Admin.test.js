import Admin from './Admin';
import { cleanup, fireEvent } from '@testing-library/react';
import '@testing-library/jest-dom';
import { renderWithProviders } from './utils/utils-for-tests';
import { screen } from '@testing-library/react';
import mockRouter from 'next-router-mock';



jest.mock('next/router', () => require('next-router-mock'));


describe('test Admin component', () => {
  afterEach(cleanup);

  test('there is the buttons', () => {
    renderWithProviders(<Admin/>);
    const btns = screen.getAllByRole('button');
    btns.forEach((item) => {
      expect(item).toBeInTheDocument();
      expect(item).toMatchSnapshot();
    })
  });

  test('there is the search input', async () => {
    renderWithProviders(<Admin/>);
    const input = screen.getByPlaceholderText('Search by name...');
    expect(input).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });

  test('extBtn returns to main page', () => {
    mockRouter.push('/');

    renderWithProviders(<Admin />);

    fireEvent.click(screen.getByText('Back'));

    expect(mockRouter).toMatchObject({
      asPath: '/',
      pathname: '/',

    });
  });
});
