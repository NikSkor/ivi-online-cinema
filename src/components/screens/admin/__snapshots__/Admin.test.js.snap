// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test Admin component there is the buttons 1`] = `
<button
  class="actionBtn searchBtn"
>
  Search
</button>
`;

exports[`test Admin component there is the buttons 2`] = `
<button
  class="actionBtn"
>
  Add
</button>
`;

exports[`test Admin component there is the buttons 3`] = `
<button
  class="actionBtn"
>
  Back
</button>
`;

exports[`test Admin component there is the search input 1`] = `
<input
  class="inputs searchInput"
  placeholder="Search by name..."
  type="text"
  value=""
/>
`;
